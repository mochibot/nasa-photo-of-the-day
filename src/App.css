.App {
  text-align: center;
}

.App-content {
  max-width: 800px;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 10px 0;
}

.App-content-image {
  width: 100%;
  margin: 20px;
}  

.App-content-video {
  margin: 20px;
}