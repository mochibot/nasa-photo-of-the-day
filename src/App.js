import React, { useState, useEffect } from "react";
import axios from 'axios';

import Content from './components/Content';
import "./App.css";

function App() {
  const [data, setData] = useState('');
  const [error, setError] = useState('');
  const [date, setDate] = useState('')

  useEffect(() => {
    axios.get(`https://api.nasa.gov/planetary/apod?api_key=DEMO_KEY${date}`)
      .then(response => {
        console.log('fetch photo success: ', response)
        setData(response.data)
      }) 
      .catch(error => {
        console.log('error fetching data: ', error)
        setError(error)
      })   
  }, [date])

  const changeDate = (event) => {
    setDate('&date=' + event.target.value);
    console.log(event.target.value);
  }

  /*
  const dateRange = [];
  for (let i = 10; i >= 1; i--) {
    let today = new Date();
    let newDate = new Date(today - i * 24 * 3600 * 1000).toISOString().slice(0, 10)
    dateRange.unshift(newDate);
  }
  dateRange.unshift('today');
  */
  let today = new Date().toISOString().slice(0, 10)

  return (
    <div className="App">
      <h1>NASA Astronomy Picture of the Day</h1>
      <input type='date' defaultValue={today} min='1995-06-16' max={today} onChange={changeDate}/>
      {/*
      <select onChange={changeDate} defaultValue={date}>
        <option disabled>Select a date</option>
        {dateRange.map(item => item === 'today' ? <option key={item} value=''>{item}</option> : <option key={item} value={item} >{item}</option>)}
      </select>
      */}
      {!data ? <div>Loading...</div> : <Content data={data}/>}
      {error && <div>Something went wrong...</div>}
    </div>
  );
}

export default App;
