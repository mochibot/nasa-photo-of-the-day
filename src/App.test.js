import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
//import AppContent from './components/AppContent'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('<App />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });
})
