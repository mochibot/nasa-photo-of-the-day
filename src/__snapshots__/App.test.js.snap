// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div
  className="App"
>
  <section
    className="ant-layout"
  >
    <header
      className="ant-layout-header"
    >
      <h2
        aria-label={null}
        className="ant-typography"
        style={
          Object {
            "WebkitLineClamp": null,
            "color": "white",
            "padding": "9px",
          }
        }
      >
        NASA Astronomy Picture of the Day
      </h2>
    </header>
    <main
      className="ant-layout-content"
    >
      <span
        aria-label={null}
        className="ant-typography"
        style={
          Object {
            "WebkitLineClamp": null,
            "fontSize": "20px",
            "lineHeight": 2.5,
          }
        }
      >
        Select another date to view more pictures
      </span>
      <br />
      <span
        className="ant-calendar-picker"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "width": "278px",
          }
        }
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
        >
          <input
            className="ant-calendar-picker-input ant-input"
            placeholder="Select date"
            readOnly={true}
            value=""
          />
          <i
            aria-label="icon: calendar"
            className="anticon anticon-calendar ant-calendar-picker-icon"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="calendar"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
              />
            </svg>
          </i>
        </div>
      </span>
      <br />
      <i
        aria-label="icon: sync"
        className="anticon anticon-sync"
        style={
          Object {
            "fontSize": 30,
            "margin": "40px",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="anticon-spin"
          data-icon="sync"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M168 504.2c1-43.7 10-86.1 26.9-126 17.3-41 42.1-77.7 73.7-109.4S337 212.3 378 195c42.4-17.9 87.4-27 133.9-27s91.5 9.1 133.8 27A341.5 341.5 0 0 1 755 268.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 0 0 3 14.1l175.7 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c0-6.7-7.7-10.5-12.9-6.3l-56.4 44.1C765.8 155.1 646.2 92 511.8 92 282.7 92 96.3 275.6 92 503.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8zm756 7.8h-60c-4.4 0-7.9 3.5-8 7.8-1 43.7-10 86.1-26.9 126-17.3 41-42.1 77.8-73.7 109.4A342.45 342.45 0 0 1 512.1 856a342.24 342.24 0 0 1-243.2-100.8c-9.9-9.9-19.2-20.4-27.8-31.4l60.2-47a8 8 0 0 0-3-14.1l-175.7-43c-5-1.2-9.9 2.6-9.9 7.7l-.7 181c0 6.7 7.7 10.5 12.9 6.3l56.4-44.1C258.2 868.9 377.8 932 512.2 932c229.2 0 415.5-183.7 419.8-411.8a8 8 0 0 0-8-8.2z"
          />
        </svg>
      </i>
      <br />
      
    </main>
    <footer
      className="ant-layout-footer"
      style={
        Object {
          "background": "#001529",
          "color": "white",
          "fontSize": "20px",
        }
      }
    >
      All images/videos obtained from 
      <a
        href="https://api.nasa.gov/index.html"
      >
        api.nasa.gov
      </a>
    </footer>
  </section>
</div>
`;
